src/main/java/com/exasol/udfstartuptimeimprover/UdfStartUpTimeImprover.java

```diff
@@ -0,0 +1,171 @@
+     * BucketFS as 'IDENTIFIED BY'.
+     */
+    public static final String PARAMETER_CONNECTION = "CONNECTION_NAME";
+    /** Name for the UDF parameter that contains the BucketFS port. */
+    public static final String PARAMETER_BUCKET_FS_PORT = "BUCKET_FS_PORT";
+    /** Name for the UDF parameter that contains the BucketFS service name */
+    public static final String PARAMETER_BUCKET_FS_SERVICE = "BUCKET_FS_SERVICE";
+    /** Name for the UDF parameter that contains the BucketFS bucket name. */
+    public static final String PARAMETER_BUCKET_FS_BUCKET = "BUCKET_FS_BUCKET";
+    /**
+     * Name for the UDF parameter that contains destination path in BucketFS where this UDF writes the class dump to.
+     */
+    public static final String PARAMETER_PATH_FOR_DUMP = "PATH_FOR_DUMP";
+    private static final String DUMP_IN_TMP = "/tmp/dump.jsa";
+
+    private UdfStartUpTimeImprover() {
+        // static class
+    }
+
+    /**
+     * Entrypoint of the startup time improver UDF.
+     *
+     * 
+     * @param exaMetadata exasol metadata
+     * @param exaIterator iterator
+     * 
+     * @throws Exception if something goes wrong
+     */
+    @SuppressWarnings({ "java:S112", "java:S1130" }) // Exception is too generic and not thrown. This signature is
+    // however given by the UDF framework
+    public static String run(final ExaMetadata exaMetadata, final ExaIterator exaIterator) throws Exception {
+        return runInternal(exaMetadata, exaIterator);
+    }
+
+    private static String runInternal(final ExaMetadata exaMetadata, final ExaIterator exaIterator) {
+        final String udfDefinitionString = readStringParameter(exaIterator, PARAMETER_UDF_DEFINITION);
+        final UdfDefinition udfDefinition = new UdfDefinitionParser().parseUdfDefinition(udfDefinitionString);
+        final Path pathToClassList = Path.of(readStringParameter(exaIterator, PARAMETER_CLASSES));
+        final String connectionName = readStringParameter(exaIterator, PARAMETER_CONNECTION);
+        final String bfsService = readStringParameter(exaIterator, PARAMETER_BUCKET_FS_SERVICE);
+        final String bfsBucket = readStringParameter(exaIterator, PARAMETER_BUCKET_FS_BUCKET);
+        final String pathForDump = readStringParameter(exaIterator, PARAMETER_PATH_FOR_DUMP);
+        final int bfsPort = readIntParameter(exaIterator, PARAMETER_BUCKET_FS_PORT);
+        final ExaConnectionInformation bucketFsConnection = getConnection(exaMetadata, connectionName);
+        assertClassListExists(pathToClassList);
+        runClassDumpGeneration(udfDefinition.getJars(), pathToClassList);
+        final UnsynchronizedBucket bucket = WriteEnabledBucket.builder().useTls(false).raiseTlsErrors(false)
+                .ipAddress("127.0.0.1").port(bfsPort).serviceName(bfsService).name(bfsBucket).readPassword("")
+                .writePassword(bucketFsConnection.getPassword()).build();
```

Comment:
```suggestion
        final UnsynchronizedBucket bucket = WriteEnabledBucket.builder().useTls(true).raiseTlsErrors(false)
                .ipAddress("127.0.0.1").port(bfsPort).serviceName(bfsService).name(bfsBucket).readPassword("")
                .writePassword(bucketFsConnection.getPassword()).build();
```

This is a local connection. But could we still use TLS?

Reply:
I'm not sure. I thought it would cause trouble with certificates.